### ISSUE TITLE
[Bug]: MO crash running mo-tester

### ISSUE THREAD

**Issue by dengn** [matrixorigin/matrixone#3159]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):e1319c52cf87b17a98008ae0dda79b72237d9ed1
- Hardware parameters:Mac M1 ARM 16GB Memory
- OS type:OSX
- Others:
```

### Actual Behavior

 Mo crashed in the middle running MO-Tester.

Some log:

```
{"level":"INFO","time":"2022/06/21 09:27:21.798192 +0800","caller":"txnbase/txnmgr.go:192","msg":"[PrepareCommit]","txns":{"duration":"33.75µs","count":1}}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816520 +0800","caller":"txnimpl/txndb.go:321","msg":"Txn-2482 ApplyCommit Takes 1.791µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.816547 +0800","caller":"txnbase/txnmgr.go:219","msg":"[Commit]","txns":{"count":1,"duration":"18.349208ms"}}
{"level":"INFO","time":"2022/06/21 09:27:21.816559 +0800","caller":"frontend/session.go:82","msg":"switch from 2 to 3"}
{"level":"INFO","time":"2022/06/21 09:27:21.816567 +0800","caller":"frontend/session.go:573","msg":"clean tae txn"}
{"level":"INFO","time":"2022/06/21 09:27:21.816571 +0800","caller":"frontend/session.go:82","msg":"switch from 3 to 0"}
{"level":"INFO","time":"2022/06/21 09:27:21.816576 +0800","caller":"frontend/routine.go:117","msg":"connection id 1001 , the time of handling the request 18.552875ms"}
{"level":"INFO","time":"2022/06/21 09:27:21.816581 +0800","caller":"frontend/mysql_cmd_executor.go:2299","msg":"cmd 3"}
{"level":"INFO","time":"2022/06/21 09:27:21.816585 +0800","caller":"frontend/mysql_cmd_executor.go:2322","msg":"query:CREATE table t1(a int, b float);\n"}
{"level":"INFO","time":"2022/06/21 09:27:21.816682 +0800","caller":"frontend/session.go:433","msg":"start txn autocommit if needed"}
{"level":"INFO","time":"2022/06/21 09:27:21.816688 +0800","caller":"frontend/session.go:369","msg":"current txn state 0"}
{"level":"INFO","time":"2022/06/21 09:27:21.816692 +0800","caller":"frontend/session.go:438","msg":"need create new txn"}
{"level":"INFO","time":"2022/06/21 09:27:21.816695 +0800","caller":"frontend/session.go:419","msg":"start txn by autocommit"}
{"level":"INFO","time":"2022/06/21 09:27:21.816701 +0800","caller":"frontend/session.go:82","msg":"switch from 0 to 2"}
{"level":"INFO","time":"2022/06/21 09:27:21.816704 +0800","caller":"frontend/mysql_cmd_executor.go:1874","msg":"start autocommit txn in default"}
{"level":"INFO","time":"2022/06/21 09:27:21.816714 +0800","caller":"frontend/mysql_cmd_executor.go:2069","msg":"time of Exec.Build : 4.208µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.816738 +0800","caller":"frontend/mysql_cmd_executor.go:2185","msg":"time of Exec.Run : 15.334µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.816770 +0800","caller":"frontend/mysql_cmd_executor.go:2212","msg":"time of SendResponse 23.25µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.816790 +0800","caller":"frontend/session.go:510","msg":"commit autocommit only"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816835 +0800","caller":"txnbase/txnmgr.go:132","msg":"[PreCommit]","txn":{"id":2483,"startTs":3735,"state":"Active"},"duration":"1.209µs"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816841 +0800","caller":"txnbase/txn.go:190","msg":"Prepare Committing 2483"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816849 +0800","caller":"txnimpl/txndb.go:357","msg":"Txn-2483 PrepareCommit Takes 583ns"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816863 +0800","caller":"txnimpl/cmdmgr.go:76","msg":"ApplyTxnRecord LSN=1253, Size=180"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.816868 +0800","caller":"txnimpl/store.go:410","msg":"Txn-2483 PrepareCommit Takes 14.541µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.816872 +0800","caller":"txnbase/txnmgr.go:192","msg":"[PrepareCommit]","txns":{"duration":"39.208µs","count":1}}
{"level":"DEBUG","time":"2022/06/21 09:27:21.835596 +0800","caller":"txnimpl/txndb.go:321","msg":"Txn-2483 ApplyCommit Takes 4.75µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.835637 +0800","caller":"txnbase/txnmgr.go:219","msg":"[Commit]","txns":{"count":1,"duration":"18.759958ms"}}
{"level":"INFO","time":"2022/06/21 09:27:21.835696 +0800","caller":"frontend/session.go:82","msg":"switch from 2 to 3"}
{"level":"INFO","time":"2022/06/21 09:27:21.835709 +0800","caller":"frontend/session.go:573","msg":"clean tae txn"}
{"level":"INFO","time":"2022/06/21 09:27:21.835713 +0800","caller":"frontend/session.go:82","msg":"switch from 3 to 0"}
{"level":"INFO","time":"2022/06/21 09:27:21.835732 +0800","caller":"frontend/routine.go:117","msg":"connection id 1001 , the time of handling the request 19.138459ms"}
{"level":"INFO","time":"2022/06/21 09:27:21.835739 +0800","caller":"frontend/mysql_cmd_executor.go:2299","msg":"cmd 3"}
{"level":"INFO","time":"2022/06/21 09:27:21.835745 +0800","caller":"frontend/mysql_cmd_executor.go:2322","msg":"query:insert into t1 select ceiling(12124.413), ceiling(..."}
{"level":"INFO","time":"2022/06/21 09:27:21.835855 +0800","caller":"frontend/session.go:433","msg":"start txn autocommit if needed"}
{"level":"INFO","time":"2022/06/21 09:27:21.835860 +0800","caller":"frontend/session.go:369","msg":"current txn state 0"}
{"level":"INFO","time":"2022/06/21 09:27:21.835863 +0800","caller":"frontend/session.go:438","msg":"need create new txn"}
{"level":"INFO","time":"2022/06/21 09:27:21.835866 +0800","caller":"frontend/session.go:419","msg":"start txn by autocommit"}
{"level":"INFO","time":"2022/06/21 09:27:21.835873 +0800","caller":"frontend/session.go:82","msg":"switch from 0 to 2"}
{"level":"INFO","time":"2022/06/21 09:27:21.835876 +0800","caller":"frontend/mysql_cmd_executor.go:1874","msg":"start autocommit txn in default"}
{"level":"INFO","time":"2022/06/21 09:27:21.835962 +0800","caller":"frontend/session.go:651","msg":"dbName func_math_ceil tableNames [t1]"}
{"level":"INFO","time":"2022/06/21 09:27:21.836005 +0800","caller":"frontend/mysql_cmd_executor.go:2069","msg":"time of Exec.Build : 123.083µs"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.836094 +0800","caller":"txnimpl/localseg.go:192","msg":"Offset=0, ToAppend=1, EstimateSize=0"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.836155 +0800","caller":"txnimpl/localseg.go:202","msg":"Appended: 1, Space:9999"}
{"level":"INFO","time":"2022/06/21 09:27:21.836165 +0800","caller":"frontend/mysql_cmd_executor.go:2185","msg":"time of Exec.Run : 110.791µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.836208 +0800","caller":"frontend/mysql_cmd_executor.go:2212","msg":"time of SendResponse 34.417µs"}
{"level":"INFO","time":"2022/06/21 09:27:21.836216 +0800","caller":"frontend/session.go:510","msg":"commit autocommit only"}
{"level":"DEBUG","time":"2022/06/21 09:27:21.836317 +0800","caller":"segmentio/driver.go:299","msg":" store/tae/1269.seg-0x1400d2b9090 | SegmentFile | Mount | Null-1-1 | Log Level1 0x1400e1dd0b0-ffffffffffffffff"}
panic: open store/tae/1269.seg: too many open files

goroutine 230 [running]:
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/dataio/segmentio.openSegment({0x1400d1839c8, 0x12}, 0x4f5)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/dataio/segmentio/segment.go:83 +0x210
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/dataio/segmentio.(*segmentFactory).Build(0x14002b5d748?, {0x140003da4a0, 0xb}, 0x50?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/dataio/segmentio/segment.go:45 +0x98
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.newSegment(0x1400ee83800, {0x10662de20?, 0x10775ed98?}, {0x106649010?, 0x14000402300}, {0x140003da4a0?, 0x14002b5d800?})
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/tables/segment.go:41 +0x58
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/tables.(*DataFactory).MakeSegmentFactory.func1(0x14002b5d798?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/tables/tables.go:52 +0x3c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.NewSegmentEntry(0x14011f61b80, {0x106653460?, 0x140031e2950}, 0x0, 0x140031e2e90)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/catalog/segment.go:58 +0x1c4
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/catalog.(*TableEntry).CreateSegment(0x14011f61b80, {0x106653460, 0x140031e2950}, 0x80?, 0x14002b5d8c8?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/catalog/table.go:154 +0x98
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).CreateSegment(0x1400e12cf00)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/table.go:167 +0x68
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*localSegment).prepareApplyNode(0x1400ebaf680, {0x106652938?, 0x1400ebaf700})
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/localseg.go:143 +0x200
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*localSegment).PrepareApply(...)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/localseg.go:127
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnTable).PreCommit(0x14002b5dae8?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/table.go:658 +0x90
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnDB).PreCommit(0x1400ee83340)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/txndb.go:332 +0xfc
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnimpl.(*txnStore).PreCommit(0x14002b5dc08?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnimpl/store.go:366 +0x7c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*Txn).PreCommit(0x10772b2e0?)
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnbase/txn.go:242 +0x2c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).onPreCommit(0x14002b5ddb8?, {0x1066532f0, 0x1400ef46240})
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:131 +0x4c
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/txn/txnbase.(*TxnManager).onPreparing(0x14000286460, {0x14000150000, 0x1, 0x0?})
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/txn/txnbase/txnmgr.go:156 +0xb4
github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start.func1()
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:83 +0x1bc
created by github.com/matrixorigin/matrixone/pkg/vm/engine/tae/logstore/sm.(*safeQueue).Start
	/Users/nandeng/Test/matrixone_new/matrixone/pkg/vm/engine/tae/logstore/sm/safeq.go:62 +0xf4
```

### Steps to Reproduce

```markdown
./run.sh for MO-Tester
```

**Comment by yingfeng** [comment]
too many open files

**Comment by LeftHandCold** [comment]
The default maxfiles of macos is 256, tae gc will not be too frequent, you need to execute the command "ulimit -n 65536"

**Comment by dengn** [comment]
ok. it works for now. But this is a workaround, eventually this issue still has to be solved.

**Comment by LeftHandCold** [comment]
Executing MO-Tester will quickly create 1000+ files, GC will not be executed frequently, so the number of files will exceed the limit of 256.

**Comment by fengttt** [comment]
First there is no bvt case associated with this one.    Second, I tried incr ulimit and indeed got better.     I still see mo-server crash from time to time but looks like different issue.   Push to 0.6.

**Comment by LeftHandCold** [comment]
> First there is no bvt case associated with this one. Second, I tried incr ulimit and indeed got better. I still see mo-server crash from time to time but looks like different issue. Push to 0.6.

Run bvt will have a large number of create&drop table operations, create a table to insert data, tae will create a segment file, so a large number of segment files will be created during operation

**Comment by LeftHandCold** [comment]
This issue has been resolved in 0.6
